docs/modules/reference/pages/configuration/history.adoc

```diff
@@ -0,0 +1,133 @@
+
+[[ga-alarm-history]]
+= Alarm History
+
+The alarm history feature integrates with Elasticsearch to provide long term storage and maintain a history of alarm state changes.
+
+When enabled, alarms are indexed in Elasticsearch when they are created, deleted, or when any of the "interesting" fields on the alarm are updated (more on this below.)
+
+Alarms are indexed so that operators can answer the following questions:
+
+. What were all the state changes of a particular alarm?
+. What was the last known state of an alarm at a given point in time?
+. Which alarms were present (i.e. not deleted) on the system at a given point in time?
+. Which alarms are currently present on the system?
+
+A simple REST API is also made available for the purposes of evaluating the results, verifying the data that is stored and providing examples on how to query the data.
+
+[[ga-alarm-history-requirements]]
+== Requirements
+
+This feature requires Elasticsearch 7.0+.
+
+[[ga-alarm-history-setup]]
+== Setup
+
+Alarm history indexing can be enabled as follows:
+
+First, log in to the Karaf shell of your {page-component-title} instance and configure the Elasticsearch client settings to point to your Elasticsearch cluster.
+See <<elasticsearch/introduction.adoc#ga-elasticsearch-integration-configuration, Elasticsearch Integration Configuration>> for a complete list of available options.
+
+[source, karaf]
+----
+$ ssh -p 8101 admin@localhost
+...
+admin@opennms()> config:edit org.opennms.features.alarms.history.elastic
+admin@opennms()> config:property-set elasticUrl http://es:9200
+admin@opennms()> config:update
+----
+
+. To make the alarm history feature start automatically on future service starts, add the line `opennms-alarm-history-elastic` to the file `$\{OPENNMS_HOME}/etc/featuresBoot.d/alarm.boot`.
+(Create the file if it does not exist.)
+. To make the alarm history feature start now, log in to the Karaf shell and run the following command to load the feature:
+
+ feature:install opennms-alarm-history-elastic
+
+[[ga-alarm-history-indexing]]
+== Alarm indexing
+
+When alarms are initially created, we push a document to Elasticsearch that includes all of the alarm fields as well as additional details on some of the related objects (i.e., the node.)
+
+In order to avoid pushing a new document every time a new event is reduced on to an existing alarm, we only push a new document when (at least) one of these conditions are met:
```

Comment:
```suggestion
To avoid pushing a new document every time a new event is reduced on to an existing alarm, we push a new document only when (at least) one of these conditions are met:
```
I'm not clear on what "reduced onto an existing alarm means".